optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -1,18 +1,18 @@
+from __future__ import annotations
+
 import abc
+from collections.abc import Callable
+from collections.abc import Container
+from collections.abc import Generator
+from collections.abc import Iterator
```

Comment:
I think most of Optuna codebase uses `typing.Callable`. Do you think these changes are necessary to fix the CI?

Reply:
This is not necessary to fix CI but just followed [PEP584](https://peps.python.org/pep-0585/) as written in https://github.com/optuna/optuna/issues/4508. If it is better to revert this part, let me know.

Reply:
Thank you for letting me know. As mentioned in [PEP585](https://peps.python.org/pep-0585/), we can keep the current changes since the corresponding classes in the `typing` are deprecated.

> collections.abc.Callable
> collections.abc.Container
> collections.abc.Generator
> collections.abc.Iterator
> Importing those from typing is deprecated.